ledger/acctonline.go

```diff
@@ -99,6 +106,14 @@ type onlineAccounts struct {
 	// baseAccounts stores the most recently used accounts, at exactly dbRound
 	baseOnlineAccounts lruOnlineAccounts
 
+	// onlineAccountsCache contains up to onlineAccountsCacheMaxSize accounts with their complete history
+	// for the range [Lastest - MaxBalLookback, Latest - lookback]
```

Comment:
What is `Latest` here? `dbRound` as written in the database?

This invariant doesn't hold I think.
1. Between after `commitRound()` and before `postCommit()` the cache is not pruned and contains elements that are too old w.r.t. the database. It's not a problem, but the invariant "lies" nevertheless.
2. In `lookupOnlineAccountData()` between lines `persistedDataHistory, _ := ao.accountsq.lookupOnlineHistory(addr)` and `for _, data := range persistedDataHistory {` the database could move ahead. As a result, the cache will miss some new elements. (this is a problem)

Reply:
> the database could move ahead. As a result, the cache will miss some new elements. (this is a problem)
if the DB advances, it is not a problem, but writing the same address into the cache in `postCommit` while is a problem because the history will not get written. `writeFront` should return a flag indicating it wrote data, and not writing here should be an error.
This means we cannot load the history without holding the `ao.accountsMu` lock.
Good catch!